spec/integ/crypto/rust-crypto.spec.ts

```diff
@@ -237,6 +247,143 @@ describe("MatrixClient.initRustCrypto", () => {
         // The final call should have progress == total == -1
         expect(progressListener).toHaveBeenLastCalledWith(-1, -1);
     }, 60000);
+
+    describe("Legacy trust migration", () => {
+        beforeEach(() => {
+            fetchMock.reset();
+        });
+
+        it("should not revert to untrusted if legacy was trusted but msk not in cache, big account", async () => {
+            // Just 404 here for the test
+            fetchMock.get("path:/_matrix/client/v3/room_keys/version", {
+                status: 404,
+                body: {
+                    errcode: "M_NOT_FOUND",
+                    error: "No backup found",
+                },
+            });
+
+            mockKeyQueryForTestMigration();
+
+            const testStoreName = "test-store";
+            await populateStore(testStoreName);
+            const cryptoStore = new IndexedDBCryptoStore(indexedDB, testStoreName);
+            await cryptoStore.startup();
+
+            // Remove the master key from the cache
+            await cryptoStore.doTxn("readwrite", [IndexedDBCryptoStore.STORE_ACCOUNT], (txn) => {
+                const objectStore = txn.objectStore("account");
+                objectStore.delete(`ssss_cache:master`);
+            });
+
+            const matrixClient = createClient({
+                baseUrl: "http://test.server",
+                userId: "@vdhtest200713:matrix.org",
+                deviceId: "KMFSTJSMLB",
+                cryptoStore,
+                pickleKey: "+1k2Ppd7HIisUY824v7JtV3/oEE4yX0TqtmNPyhaD7o",
+            });
+
+            await matrixClient.initRustCrypto();
+
+            const verificationStatus = await matrixClient
+                .getCrypto()!
+                .getUserVerificationStatus("@vdhtest200713:matrix.org");
+
+            expect(verificationStatus.isCrossSigningVerified()).toBe(true);
+        }, 60000);
+
+        it("should not revert to untrusted if legacy was trusted but msk not in cache", async () => {
+            fetchMock.get("path:/_matrix/client/v3/room_keys/version", MIGRATION_BACKUP);
```

Comment:
Why do we need a backup here, but not in the other case? Can't we just return a 404 here?

Reply:
We could, it was just to have a test trying with a 404 to see if it was working.